src/raw/unix.rs

```diff
@@ -414,6 +418,32 @@ impl Socket {
     }
 
 
+    ///Sets whether this socket will be inherited to child porcesses or not.
+    pub fn set_inheritable(&self, value: bool) -> io::Result<()> {
+        // Some (or possibly all?) OS's support the `FIOCLEX` and `FIONCLEX`
+        // `ioctl`s instead, however there is no support for that in `libc`
+        // currently and no usable documentation for figuring out who supports
+        // this feature online either
+        unsafe {
+            let mut flags: libc::c_int = libc::fcntl(self.inner, libc::F_GETFD);
+            if flags < 0 {
+                return Err(io::Error::last_os_error());
+            }
+            
+            if value == true {

```

Comment:
If the fd is inheritable, then it should not be *CLosed On EXEC(2)*, so this code should be correct.